src/PrestaShopBundle/Resources/config/services/core/grid/doctrine_query_builder.yml

```diff
@@ -530,3 +530,13 @@ services:
       $languageId: '@=service("prestashop.adapter.legacy.context").getContext().language.id'
       $connection: '@doctrine.dbal.default_connection'
       $dbPrefix: '%database_prefix%'
+
+  prestashop.core.grid.query.builder.cart:
+    class: 'PrestaShop\PrestaShop\Core\Grid\Query\CartQueryBuilder'
+    parent: 'prestashop.core.grid.abstract_query_builder'
+    arguments:
+      - '@prestashop.core.query.doctrine_search_criteria_applicator'
+      - '@prestashop.adapter.shop.context'
+      - "@=service('prestashop.adapter.legacy.context').getContext().shop.id"
+      - "@=service('prestashop.adapter.legacy.context').getContext().shop.getContextShopGroupId()"
```

Comment:
there is a better way.
Shop ids can be injected into search criteria (by extending `PrestaShop\PrestaShop\Core\Search\ShopFilters` it will have the ShopConstraint inside. There is example how I did it with FeatureQueryBuilder in develop branch already.
But I don't require this change now, Im fine with this for now, and it can be improved later :v:

Reply:
Oh, it can be nice to do this improvement now 😄

Reply:
as you wish (just the PR might expand with some unexpected changes, but im ok with that :smile:)